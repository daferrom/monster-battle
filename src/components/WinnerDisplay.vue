<template>
  <div class="winner-box">
    <p class="message-winner">{{ message }} wins!</p>
  </div>
</template>

<script lang="ts">
import Vue from "vue";

export default Vue.extend({
  name: "WinnerDisplay",
  props: {
    message: {
      type: String,
      required: false,
    },
  },
});
</script>

<style lang="scss" scoped>
@import "@/styles/variables";

.winner-box {
  background: $winner-background;
  border: 1px solid $black;
  box-shadow: $box-shadow;
  border-radius: 4px;
  padding: 17px 28px;
  align-content: center;
}

.message-winner {
  font-style: normal;
  font-weight: 400;
  font-size: 22px;
  line-height: 26px;
}
</style>
