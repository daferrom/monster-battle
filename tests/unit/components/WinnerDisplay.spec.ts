import { mount } from "@vue/test-utils";
import WinnerDisplay from "@/components/WinnerDisplay.vue";
import Vuetify from "vuetify";

const vuetify = new Vuetify();

describe("WinnerDisplay Component", () => {
  test("should match with the snapshot", () => {
    const wrapper = mount(WinnerDisplay, {
      vuetify,
      propsData: {
        message: "",
      },
    });

    expect(wrapper.html()).toMatchSnapshot();
  });

  test("should show the message Dead Unicorn wins!", () => {
    const msg = "Dead Unicorn";
    const wrapper = mount(WinnerDisplay, {
      vuetify,
      propsData: {
        message: msg,
      },
    });

    expect(wrapper.text()).toContain(msg);
    expect(wrapper.classes("winner-box")).toBeTruthy();
  });
});
