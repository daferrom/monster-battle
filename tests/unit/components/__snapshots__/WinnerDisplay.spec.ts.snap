// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WinnerDisplay Component should match with the snapshot 1`] = `
<div class="winner-box">
  <p class="message-winner"> wins!</p>
</div>
`;
